pyadjoint/checkpointing.py

```diff
@@ -78,9 +88,15 @@ def __init__(self, schedule, tape):
         self.forward_schedule.append(self._current_action)
         # Tell the tape to only checkpoint input data until told otherwise.
         self.tape.latest_checkpoint = 0
-        self.end_timestep(-1)
         self._keep_init_state_in_work = False
-        self._adj_deps_cleaned = False
+        self._gc_timestep_frequency = gc_timestep_frequency
+        self._gc_generation = gc_generation
+        # ``self._global_deps`` stores checkpoint dependencies that remain unchanged across
+        # timesteps (``self.tape.timesteps``). During the forward taping process, the code
+        # checks whether a dependency is in ``self._global_deps`` to avoid unnecessary clearing
+        # and recreation of its checkpoint data.
```

Comment:
@connorjward Here, I try to explain why ``self._global_deps`` and what is going to store.